heka/plugins/s3splitfile/s3splitfile_input.go

```diff
@@ -224,6 +231,13 @@ func (input *S3SplitFileInput) fetcher(runner pipeline.InputRunner, wg *sync.Wai
 			}
 			duration = time.Now().UTC().Sub(startTime).Seconds()
 			runner.LogMessage(fmt.Sprintf("Successfully fetched %s in %.2fs ", s3Key, duration))
+		case <-input.stop:
+			for _ = range input.listChan {
+				// drain the channel without processing the files
+				// technically the S3Iterator can still add one back on to the
+				// channel but this ensures there is room so it won't block
```

Comment:
nit: capitalize "Drain" and "Technically", add a period after "files" and "block"